tests/unit/nupic/algorithms/knn_classifier_test.py

```diff
@@ -30,6 +30,95 @@
 class KNNClassifierTest(unittest.TestCase):
 
 
+  def testDistanceMetrics(self):
+    classifier = KNNClassifier(distanceMethod="norm", distanceNorm=2.0)
+  
+    dimensionality = 40
+    protoA = np.array([0, 1, 3, 7, 11], dtype=np.int32)
+    protoB = np.array([20, 28, 30], dtype=np.int32)
+    
+    classifier.learn(protoA, 0, isSparse=dimensionality)
+    classifier.learn(protoB, 0, isSparse=dimensionality)
+    
+    # input is an arbitrary point, close to protoA, orthogonal to protoB
+    input = np.zeros(dimensionality)
+    input[:4] = 1.0
+    # input0 is used to test that the distance from a point to itself is 0
+    input0 = np.zeros(dimensionality)
+    input0[protoA] = 1.0
+
+    # Test l2 norm metric
+    _, _, dist, _ = classifier.infer(input)
+    l2Distances = [0.65465367,  1.0]
+    for actual, predicted in zip(l2Distances, dist):
+      self.assertAlmostEqual(actual, predicted, places=5,
+        msg="l2 distance norm is not calculated as expected.")
```

Comment:
If you want to break like this then start the first argument on a new line. See [PEP-8](https://www.python.org/dev/peps/pep-0008/#indentation).